packages/opentelemetry-types/src/metrics/counter.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface CounterTimeseries {
+  // Adds the given value to the current value. Values cannot be negative.
+  add(value: number): void;
+
+  // Sets the given value. Value must be larger than the current recorded value.
+  set(value: number): void;
```

Comment:
Should these be `increment` and `decrement` instead? It would make more sense for a counter.

Reply:
Yeah probably. I'm also curious where the `Values cannot be negative.` requirement comes from.

Reply:
I would think that fundamentally a "counter" represents counting something, which intuitively can't be negative. If a value can be negative, then to me it is not really a count of something tangible but rather represents something subtly different. For example, what does it mean that a server served -20 requests?

On the other hand, I could see something like "available CPU-seconds" being negative if the process is in debt to its scheduler. But that metric type I would say is a gauge and not a counter.

Reply:
Side note, based on the rfcs it looks like counters only have `add()` and gauges only have `set()` now. There are options to allow negative options for counters or indicate that a gauge only increases in value.

Reply:
OK, interesting - these are squishy concepts in some ways. Within the Stackdriver Monitoring product, counters (called "cumulative" in our API) have the interesting difference from gauges, in that you can take the rate of a cumulative but can't for a gauge.